layers/cars.py

```diff
@@ -0,0 +1,153 @@
+import geopandas as gpd
+import pandas as pd
+import numpy as np
+import subprocess
+import datetime
+import os
+
+#Code from https://github.com/dabreegster/popgetter/tree/main
+def _last_update(file_path):
+    """
+    Returns the date and time of the last update to the file at `file_path`.
+    """
+    if not os.path.exists(file_path):
+        return None
+    last_update = os.path.getmtime(file_path)
+    return datetime.datetime.fromtimestamp(last_update)
+
+
+#Code from https://github.com/dabreegster/popgetter/tree/main
+def download_from_arcgis_online(serviceItemId, output_file, force=False):
+    """
+    Downloads data from ArcGIS Online and saves it to a file (`output_file`). This function can only download data that is available to anonymous users.
+    The data will only be downloaded if the output file does not exist, or if the data on ArcGIS Online has been updated since the output file was last updated. Use `force=True` will cause the data to be re-downloaded if it an uptodate file exists locally.
+    """
+    try:
+        from arcgis.gis import GIS
+    except ImportError:
+        print("Unable to import `arcgis`. Please install the `arcgis` package, using the command `pip install requirements-non-foss.txt.")
+        return
+    
+    # Anonymous access to ArcGIS Online
+    gis = GIS()
+
+    # Get the `Item`, then, `FeatureLayer` then 'FeatureSet`:
+    agol_item = gis.content.get(serviceItemId)
+    print(f"Got item: {agol_item}")
+    print(f"item metadata: {agol_item.metadata}")
+
+    agol_layer = agol_item.layers[0]
+
+    # Get the last edit datetime for the layer
+    # print(f"Got layer: {agol_layer.properties}")
+    lyr_props = agol_layer.properties
+    # Epoch time in milliseconds - convert to datetime
+    lyr_last_edit = lyr_props.get("editingInfo", {}).get("lastEditDate", None)
+    if lyr_last_edit:
+        lyr_last_edit = datetime.datetime.fromtimestamp(lyr_last_edit/1000)
+
+    print(f"last_edit: {lyr_last_edit}")
+
+    # If the output file exists, check the last edit time
+    output_last_edit = _last_update(output_file)
+
+    if not force and output_last_edit and lyr_last_edit and output_last_edit > lyr_last_edit:
+        print(f"Output file is up-to-date: {output_file}")
+        return
+    
+    print(f"Output file is out-of-date: {output_file}")
+
+    agol_feature_set = agol_layer.query()
+    print(f"Got feature set: {len(agol_feature_set)}")
+    
+    # Write to geojson file
+    with open(output_file, "w") as f:
+        f.write(agol_feature_set.to_geojson)
+
+    print("Done")
+
+
+#Code from https://github.com/dabreegster/popgetter/tree/main
+def download_vehicle_ownership(census_url, working_dir):
```

Comment:
This is so good to see, nice function and sure it could be adapted for other input datasets :rocket: